File: python-wrapper/src/python/model/FrontEndMigration.py
import json
from typing import Optional, List, Union, Dict
from model.embed.FrontEndMigrationStatus import FrontEndMigrationStatus
from model.PropertyStub import DecryptedPropertyStub
from dataclasses import field, dataclass

@dataclass
class FrontEndMigration:
	id: str
	rev: Optional[str] = None
	deletion_date: Optional[int] = None
	name: Optional[str] = None
	start_date: Optional[int] = None
	end_date: Optional[int] = None
	status: Optional['FrontEndMigrationStatus'] = None
	logs: Optional[str] = None
	user_id: Optional[str] = None
	start_key: Optional[str] = None
	start_key_doc_id: Optional[str] = None
	process_count: Optional[int] = None
	properties: List['DecryptedPropertyStub'] = field(default_factory=list)

	def __serialize__(self) -> object:
		return {
			"id": self.id,
			"rev": self.rev,
			"deletionDate": self.deletion_date,
			"name": self.name,
			"startDate": self.start_date,
			"endDate": self.end_date,
			"status": self.status.__serialize__() if self.status is not None else None,
			"logs": self.logs,
			"userId": self.user_id,
			"startKey": self.start_key,
			"startKeyDocId": self.start_key_doc_id,
			"processCount": self.process_count,
			"properties": [x0.__serialize__() for x0 in self.properties],
		}

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'FrontEndMigration':
		deserialized_dict: dict[str, object]
		if isinstance(data, str):
			deserialized_dict = json.loads(data)
		else:
			deserialized_dict = data
		return cls(
			id = deserialized_dict["id"],
			rev = deserialized_dict.get("rev"),
			deletion_date = deserialized_dict.get("deletionDate"),
			name = deserialized_dict.get("name"),
			start_date = deserialized_dict.get("startDate"),
			end_date = deserialized_dict.get("endDate"),
			status = FrontEndMigrationStatus._deserialize(deserialized_dict.get("status")) if deserialized_dict.get("status") is not None else None,
			logs = deserialized_dict.get("logs"),
			user_id = deserialized_dict.get("userId"),
			start_key = deserialized_dict.get("startKey"),
			start_key_doc_id = deserialized_dict.get("startKeyDocId"),
			process_count = deserialized_dict.get("processCount"),
			properties = [DecryptedPropertyStub._deserialize(x0) for x0 in deserialized_dict["properties"]],
		)

__all__ = ['FrontEndMigration']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/FrontEndMigration.kt
package com.icure.cardinal.sdk.model

import com.icure.cardinal.sdk.model.base.StoredDocument
import com.icure.cardinal.sdk.model.embed.FrontEndMigrationStatus
import com.icure.cardinal.sdk.utils.DefaultValue
import kotlinx.serialization.Serializable
import kotlin.Long
import kotlin.String
import kotlin.collections.Set

@Serializable
public data class FrontEndMigration(
	override val id: String,
	override val rev: String? = null,
	override val deletionDate: Long? = null,
	public val name: String? = null,
	public val startDate: Long? = null,
	public val endDate: Long? = null,
	public val status: FrontEndMigrationStatus? = null,
	public val logs: String? = null,
	public val userId: String? = null,
	public val startKey: String? = null,
	public val startKeyDocId: String? = null,
	public val processCount: Long? = null,
	@DefaultValue("emptySet()")
	public val properties: Set<DecryptedPropertyStub> = emptySet(),
) : StoredDocument
